### PR TITLE
Adding Timestamp Check to Join Analyzer

### PR DESCRIPTION
**Pull request by pkundurthy**
## Summary

- This PR adds checks on timestamps in a Join source
   - we run the exact same checks as added in this PR:
   - https://github.com/airbnb/chronon/pull/794

The Analyzer class' method analyzeJoin will run the mentioned checks all the time.
Checks will only run when ts column is provided.

## Why / Goal

For Joins with an incorrect timestamp field is provided (for e.g. UNIX_TIMESTAMP(ts_start, 'yyyy-MM-dd') -> NULL), this option will ensure that we catch such cases.

## Test Plan

- [x] Added Unit Tests
- [ ] Covered by existing CI
- [x] Integration tested

** successful run ** 
```
[2024-08-06 17:07:06] {Analyzer:330} - ANALYSIS TIMESTAMP completed for join/joins/help_center.help_center_article_browse_user.v1.
check notNullCount: 1000
check badRangeCount: 0
```

** "CAST(NULL AS BIGINT)" **
```
Exception in thread "main" java.lang.AssertionError: assertion failed: [ERROR]: Join validation failed.
 Please check that left source has non-null timestamps.
 check notNullCount: 0
```

** "ts_unixtime*1000" **
```
Exception in thread "main" java.lang.AssertionError: assertion failed: [ERROR]: Join validation failed.
 Please check that left source has valid epoch millisecond timestamps.
 badRangeCount: 1000
```

** integration test post refactor copy-paste code **
```
[2024-08-06 21:45:55] {Analyzer:561} - ANALYSIS TIMESTAMP completed for group_by/ispp_user_recent_reservations_guest_reservations_v1ispp.user_recent_reservations.guest_reservations_v1.
check notNullCount: 1000
check badRangeCount: 0
```

## Checklist
- [x] Documentation update

### CODE REVIEW

**Review by pengyu-hou — APPROVED** [review]
LGTM

**Review by pengyu-hou — APPROVED** [review]
LGTM. Thanks for the change

File: spark/src/main/scala/ai/chronon/spark/Analyzer.scala [hunk 1]
```diff
@@ -302,6 +302,35 @@ class Analyzer(tableUtils: TableUtils,
...
+      logger.info(s"""ANALYSIS TIMESTAMP completed for join/${name}.
+                     |check notNullCount: ${timestampChecks("notNullCount")}
+                     |check badRangeCount: ${timestampChecks("badRangeCount")}
+                     |""".stripMargin)
+
+    } else {
+      logger.info(s"""ANALYSIS TIMESTAMP completed for join/${name}.
+                     |check TsColumn: ${timestampChecks("noTsColumn")}
+                     |""".stripMargin)
+    }
```
Comment (donghanz) [inline spark/src/main/scala/ai/chronon/spark/Analyzer.scala:332]: nit: I think this validation part can either be integrated into `runTimestampChecks()` or be refactored as an separate method, say `validateTimestamChecks(timestampChecks, "join")`

Comment (pkundurthy) [reply spark/src/main/scala/ai/chronon/spark/Analyzer.scala]: done.

**Review by hzding621 — APPROVED** [review]

**Pull request merged**